snr.py

```diff
@@ -0,0 +1,139 @@
+import numpy as np
+import astropy.units as u
+
+
+def snr(counts, npix=1, nb=np.inf, background=0, darkcurrent=0,
+        readnoise=0, gain=0, ADerr=0.289):
+    """ A function to calculate the signal to noise ratio (SNR) of an
+    astronomical observation.
+
+    Parameters
+    ----------
+    counts (float or int)
+        Total number of counts in an arbitrary exposure time.
+    npix (int)
+        Number of pixels under consideration for the signal. Default is 1.
+    nb (int)
+        Number of pixels used in the background estimation. Default
+        is set to infinite such that there is no contribution of
+        error due to background estimation. This assumes that the
+        nb will be >> npix.
+    background (int)
+        Total photons per pixel due to the backround/sky. Default is 0.
+    darkcurrent (int)
+        Total electrons per pixel due to the dark current. Default is 0.
+    readnoise (int)
+        Electrons per pixel from the read noise. Default is 0.
+    gain (int)
+        Gain of the CCD in electrions/ADU. Default is 0 such that the
+        contribution to the error due to the gain is assumed to be small.
+    ADerr (float)
+        An estimate of the 1 sigma error within the A/D converter.
+        Default is set to 0.289 (Merline & Howell, 1995).
+
+    Returns
+    -------
+    float
+    The signal to noise ratio of the given observation in sqrt(e-).
+    """
+    args = locals()  # maps arguments into a dictionary
```

Comment:
This is a matter of taste I suppose, but I'm still big fan of using the argument names directly, rather than manipulating this `locals()` dictionary. In my opinion it makes code easier to read and more transparent. Opinions @pllim @eteq?

Reply:
Using the `@u.quantity_input` decorator got rid of my need for `locals()` anyhow